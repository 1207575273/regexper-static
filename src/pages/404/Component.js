import React from 'react';
import PropTypes from 'prop-types';
import { translate, Trans } from 'react-i18next';

import Message from 'components/Message';
import AlertIcon from 'feather-icons/dist/icons/alert-octagon.svg';
import Header from 'components/Header';
import Footer from 'components/Footer';

const Component = ({ t }) => (
  <React.Fragment>
    <Header/>
    <Message type="error" icon={ AlertIcon } heading={ t('404 Page Not Found') }>
      <p><Trans>The page you have requested could not be found</Trans></p>
    </Message>
    <Footer/>
  </React.Fragment>
);

Component.propTypes = {
  t: PropTypes.func
};

export default translate()(Component);
