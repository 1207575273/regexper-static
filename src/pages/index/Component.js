import React from 'react';

import Message from 'components/Message';
import AlertIcon from 'feather-icons/dist/icons/alert-octagon.svg';
import Header from 'components/Header';
import Footer from 'components/Footer';
import App from 'components/App';

const Component = () => (
  <React.Fragment>
    <Header/>
    <noscript>
      <Message type="error" icon={ AlertIcon } heading="JavaScript Required">
        <p>You need to enable JavaScript to use Regexper.</p>
        <p>If you have concerns regarding the use of tracking code on Regexper, please see the <a href="/privacy.html">Privacy Policy</a>.</p>
      </Message>
    </noscript>
    <App/>
    <Footer/>
  </React.Fragment>
);

export default Component;
