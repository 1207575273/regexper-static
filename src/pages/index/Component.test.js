import React from 'react';
import { shallow } from 'enzyme';

import Component from './Component';

describe('Index page component', () => {
  test('rendering', () => {
    const component = shallow(
      <Component/>
    );
    expect(component).toMatchSnapshot();
  });
});
