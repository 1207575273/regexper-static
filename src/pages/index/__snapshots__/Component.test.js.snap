// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page component rendering 1`] = `
<React.Fragment>
  <Translate(Header) />
  <noscript>
    <Message
      heading="JavaScript Required"
      icon={[Function]}
      type="error"
    >
      <p>
        You need to enable JavaScript to use Regexper.
      </p>
      <p>
        If you have concerns regarding the use of tracking code on Regexper, please see the 
        <a
          href="/privacy.html"
        >
          Privacy Policy
        </a>
        .
      </p>
    </Message>
  </noscript>
  <Translate(App) />
  <Translate(Footer) />
</React.Fragment>
`;
